x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/pipeline_processors_editor_item/pipeline_processors_editor_item.tsx

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { FunctionComponent, memo, useState } from 'react';
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { ProcessorInternal, ProcessorSelector } from '../../types';
+
+import { usePipelineProcessorsContext } from '../../context';
+
+import './pipeline_processors_editor_item.scss';
+
+import { InlineTextInput } from './inline_text_input';
+
+export interface Handlers {
+  onMove: () => void;
+  onCancelMove: () => void;
+}
+
+export interface Props {
+  processor: ProcessorInternal;
+  selected: boolean;
+  handlers: Handlers;
+  selector: ProcessorSelector;
+  description?: string;
+}
+
+export const PipelineProcessorsEditorItem: FunctionComponent<Props> = memo(
```

Comment:
This is a pretty long component. Would it be worth breaking it up a little, to make it easier to maintain later on?